geoapps_utils/numerical.py

```diff
@@ -25,11 +23,11 @@ class DetectionParameters(BaseModel):
     :param max_distance: Maximum distance between points in a curve.
     """
 
-    azimuth: Optional[float] = None
-    azimuth_tol: Optional[float] = None
+    azimuth: float | None = None
+    azimuth_tol: float | None = None
     damping: float = 0
     min_edges: int = 1
-    max_distance: Optional[float] = None
+    max_distance: float | None = None
```

Comment:
Oh yea now i get what you are saying. Feel free to move this back to curve-apps on the same PR, since you have one open on each.